barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.test.cpp

```diff
@@ -221,4 +222,122 @@ TYPED_TEST(ShpleminiTest, CorrectnessOfGeminiClaimBatching)
 
     EXPECT_EQ(shplemini_result, expected_result);
 }
+
+/**
+ * @brief Libra masking univariates are used in sumcheck to prevent the leakage of witness data through the evaluations
+ * of round univariates. Here we test the opening of log_n Libra masking univariates batched with the opening of several
+ * prover polynomials and their shifts.
+ *
+ */
+TYPED_TEST(ShpleminiTest, ShpleminiWithMaskingLibraUnivariates)
+{
+    using ShpleminiProver = ShpleminiProver_<TypeParam>;
+    using ShpleminiVerifier = ShpleminiVerifier_<TypeParam>;
+    using KZG = KZG<TypeParam>;
+    using IPA = IPA<TypeParam>;
+    using Fr = typename TypeParam::ScalarField;
+    using Commitment = typename TypeParam::AffineElement;
+    using Polynomial = typename bb::Polynomial<Fr>;
+
+    const size_t n = 16;
+    const size_t log_n = 4;
+    const size_t LIBRA_UNIVARIATE_LENGTH = 12;
+
+    std::array<Fr, LIBRA_UNIVARIATE_LENGTH> interpolation_domain;
+    for (size_t idx = 0; idx < LIBRA_UNIVARIATE_LENGTH; idx++) {
+        interpolation_domain[idx] = Fr(idx);
+    }
+    // Generate multilinear polynomials, their commitments (genuine and mocked) and evaluations (genuine) at a
+    // random point.
+    auto mle_opening_point = this->random_evaluation_point(log_n); // sometimes denoted 'u'
+    auto poly1 = Polynomial::random(n);
+    auto poly2 = Polynomial::random(n, 1);
+    auto poly3 = Polynomial::random(n, 1);
+    auto poly4 = Polynomial::random(n);
+
+    std::vector<bb::Univariate<Fr, LIBRA_UNIVARIATE_LENGTH>> libra_univariates;
+    std::vector<Commitment> libra_commitments;
+    std::vector<Fr> libra_evaluations;
+    for (size_t idx = 0; idx < log_n; idx++) {
+        // generate random polynomial
+        Polynomial libra_polynomial = Polynomial::random(LIBRA_UNIVARIATE_LENGTH);
+        // create a univariate with the same coefficients (to store an array intead of a vector)
+        bb::Univariate<Fr, LIBRA_UNIVARIATE_LENGTH> libra_univariate;
+        for (size_t i = 0; i < LIBRA_UNIVARIATE_LENGTH; i++) {
```

Comment:
FYI it looks like Univariate has it's own `random_element()` method - might be cleaner

Reply:
thanks! it's ugly here: Univariates are in the Lagrange basis, Polynomials are in the monomial, which means we can't commit to a Univariate without computing its coefficients in the monomial basis or computing the Lagrange basis using SRS + I want to store the masking data as a std::vector<Univariate<FF, 12>> (could/will be turned into std::array) instead of std::vector<Polynomial>